huaweicloud/services/meeting/resource_huaweicloud_meeting_conferences.go

```diff
@@ -0,0 +1,1145 @@
+		Language:               d.Get("language").(string),
+		TimeZoneID:             strconv.Itoa(d.Get("timezone_id").(int)),
+		RecordType:             utils.Int(d.Get("record_type").(int)),
+		LiveAddress:            d.Get("live_address").(string),
+		AuxAddress:             d.Get("aux_address").(string),
+		RecordAuxStream:        utils.Int(d.Get("is_record_aux_stream").(int)),
+		RecordAuthType:         utils.Int(d.Get("record_auth_type").(int)),
+		ConcurrentParticipants: utils.Int(d.Get("participant_number").(int)),
+		Participants:           buildConferenceParticipants(d.Get("participant").([]interface{})),
+		CycleParams:            buildConferenceCycleParams(d.Get("cycle_params").([]interface{})),
+		Configuration:          buildConferenceConfiguration(d.Get("configuration").([]interface{})),
+		// The authorization token
+		Token: token,
+	}
+	if warRoomId, ok := d.GetOk("meeting_room_id"); ok {
+		opt.VmrFlag = utils.Int(1)
+		opt.VmrId = warRoomId.(string)
+	}
+
+	return opt
+}
+
+func resourceConferenceUpdate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	conf := meta.(*config.Config)
+	token, err := GetMeetingToken(conf, d)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	_, conferenceId, err := analyseIdString(d.Id())
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	opt := buildConferenceGetOpts(conferenceId, d.Get("user_id").(string), token)
+	log.Printf("[DEBUG][Before Update] The GetOpts of the cloud conference is: %v", opt)
+	client := GetMeetingV1Client(conf)
+
+	switch d.Get("status").(string) {
+	case string(ConferenceStatusUsing):
+		if d.HasChangeExcept("length") {
+			return diag.Diagnostics{
+				{
+					Severity: diag.Warning,
+					Summary:  "Update Not Supported",
+					Detail: "When the conference is in progress, the parameters except the conference length do not " +
+						"support changing.",
+				},
+			}
+		}
+		oldLen, newLen := d.GetChange("length")
+		if newLen.(int) < oldLen.(int) {
+			return diag.Diagnostics{
+				{
+					Severity: diag.Warning,
+					Summary:  "Invalid Duration",
+					Detail:   "When the conference is in progress, only extension is supported, not shortening.",
+				},
+			}
+		}
+		newDuration := newLen.(int) - oldLen.(int)
+		err = ExtendConference(client, conferenceId, getChairPassword(d), newDuration)
+		if err != nil {
+			return diag.FromErr(err)
+		}
+		return resourceConferenceRead(ctx, d, meta)
+	case string(ConferenceStatusSchedule):
+		opt := buildConferenceUpdateOpts(d, conferenceId, token)
+		resp, err := conferences.Update(client, opt)
+		if err != nil {
+			return diag.Errorf("error creating cloud conference: %s", err)
+		}
+		if len(resp) < 1 {
+			return diag.Diagnostics{
+				{
+					Severity: diag.Warning,
+					Summary:  "Internal Service Error",
+					Detail: "The corresponding cloud conference is not found after the creation is complete, please " +
+						"contact customer service to solve this problem.",
+				},
+			}
+		}
```

Comment:
I think we don't need to check the resp here

Reply:
done.